starlette/requests.py

```diff
@@ -250,8 +251,13 @@ async def form(self) -> FormData:
             content_type_header = self.headers.get("Content-Type")
             content_type, options = parse_options_header(content_type_header)
             if content_type == b"multipart/form-data":
-                multipart_parser = MultiPartParser(self.headers, self.stream())
-                self._form = await multipart_parser.parse()
+                try:
+                    multipart_parser = MultiPartParser(self.headers, self.stream())
+                    self._form = await multipart_parser.parse()
+                except MultiPartException as exc:
+                    if "app" in self.scope:
```

Comment:
https://github.com/encode/starlette/blob/master/starlette/routing.py#L253-L258

Reply:
Yep, we use the pattern in the code, as @adriangb pointed out.

Always coerce to an `HTTPException` is an option. I opted to not do it because we only use `HTTPException`s when we are in `Starlette` (as application) e.g. `"app" in self.scope`.

Reply:
Realistically, I think we could remove that pattern everywhere. IMO it's not worth supporting (nice) usage of Starlette's Request object outside of a Starlette app. If someone wants to do that, they can handle the HTTPException.

Reply:
>  If someone wants to do that, they can handle the HTTPException.

You have a point, but I think that's another discussion. I really don't think it's a burden to maintain that, but I wonder if someone really uses that code outside `Starlette` (app)... 🤔